sprint-1 release

lpendem opened the pull request:
This release PR includes these completed stories:
2: As a user I would like to know what species and size of trees can be harvested in Mt Hood Nat Forest?
6: As a user I would like to know where the harvesting areas are in Mt. Hood?

lauraGgit commented on frontend/src/app/trees/forests/tree-guidelines/wizard-view.component.html:
```diff
@@ -0,0 +1,100 @@
+<section class="usa-section">
```
My understanding was that we were not moving forward with the wizard view, is my understanding not correct?

lpendem replied:
@lauraGgit We might want to use this for the actual permit application, but we'll remove it for now.

lauraGgit replied:
@lpendem can you please link to the story that gets created for that?

lauraGgit commented on frontend/src/app/trees/trees.module.ts:
```diff
@@ -0,0 +1,65 @@
+import { TreeLocationsAllowedComponent } from './forests/tree-guidelines/tree-locations/tree-locations-allowed/tree-locations-allowed.component';
+import { TreeLocationsMapsComponent } from './forests/tree-guidelines/tree-locations/tree-locations-maps/tree-locations-maps.component';
+
+@NgModule({
```
See my general PR comment. While I think compontentization makes a ton of sense for form and resuable design assets, I don't wonder if it is overkill for the mostly content pages.

lpendem replied:
@lauraGgit agree with you. Let us revisit this in later sprints once all the content is finalized, and on pages. We also need to think of generating permit PDF with required elements.

lauraGgit commented on frontend/src/environments/environment.trees.ts:
```diff
@@ -0,0 +1,6 @@
+export const environment = {
+  production: true,
+  apiUrl: 'https://fs-intake-api-trees.app.cloud.gov/',
+  buildDate: 'Mon Jun 12 2017 22:17:33 GMT-0500 (CDT)',
```
Do the build dates & versions actually get populated on the build?

lpendem replied:
updated this to use `new Date()`.

lpendem replied:
@lauraGgit replace.build.js script currently replaces date to the current date during the deployment. However, that works on environment.`prod` with intake but not on `trees` deployment. Let us discuss or update this once intake is complete.

lauraGgit commented on frontend/src/sass/core/_variables.scss:
```diff
@@ -1,10 +1,12 @@
-$white: #FFFFFF;
+$light-grey: #cccccc;
+$white: #ffffff;
+$fs-yellow: #FEC33C;
```
This color is not being used anywhere else, I recommend removing.

lpendem replied:
@lauraGgit We are using fs-yellow in sprint-2, not sure about light-grey but we'll clean it up.

lauraGgit replied:
@lpendem again, why isn't that just part of the existing color set on the /styleguide page?

lpendem replied:
@lauraGgit this has been removed, and using existing colors already available.

lauraGgit commented on frontend/src/sass/pages/_tree-guidelines.scss:
```diff
@@ -0,0 +1,67 @@
+}
+
+.wizard-step-footer {
+  background: $light-grey;
```
Is there any reason an existing color from the https://forest-service-trees-staging.app.cloud.gov/style-guide

lpendem replied:
we are getting rid of wizard related code

lauraGgit replied:
@lpendem is there a story for that?

lauraGgit commented on server/Procfile:
```diff
@@ -1 +1 @@
-web: npm run migrate && npm run start
+web: npm run migrate && npm run undoAllSeed && npm run seed && npm run start
```
Do we want to consider moving this over to `yarn` commands?

lpendem replied:
@lauraGgit sure, we'll give it a try in next sprint.

lauraGgit replied:
@lpendem please drop the story that is created here.

lpendem replied:
@lauraGgit sure.

lauraGgit commented on server/dba/migrations/20-forests.js:
```diff
@@ -0,0 +1,22 @@
+    return queryInterface.createTable('forests', {
+      id: { type: Sequelize.INTEGER, primaryKey: true, field: 'id' },
+      forestName: { type: Sequelize.STRING(50), field: 'forest_name' },
+      description: { type: Sequelize.STRING(1000), field: 'description' },
```
I don't see where this description is on the frontend?

lpendem replied:
@lauraGgit correct, description field not being used. We'll drop once we are set with all the content.

lauraGgit replied:
@lpendem - could you please let me know what story/ task that action has been documented in?

lpendem replied:
@lauraGgit this filed is being using to store forest key works, forestName, city, state etc in story: https://github.com/nciinc/fs-intake-module/issues/5

lauraGgit commented on server/dba/migrations/20-forests.js:
```diff
@@ -0,0 +1,22 @@
+    });
+  },
+  down: function(queryInterface, Sequelize) {
+    return queryInterface.dropTable('forests');
```
Why are there dropTables being used instead of altering the columns. Please take a look at the earlier migrations and the `doTransaction` method that promises multiple changes.

lauraGgit replied:
@lpendem ok, please drop the story link when complete.

lpendem replied:
@lauraGgit here it is: https://github.com/nciinc/fs-intake-module/issues/78

lauraGgit commented on server/dba/seeders/prod/01-forests.js:
```diff
@@ -0,0 +1,61 @@
+'use strict';
```
this should be updated to reflect that is just for christmas trees as this information will not be useful to other permit types. We may want to consider changing the IDs to be the actual national service IDs which we can provide for you.

lpendem replied:
@lauraGgit we have added `org_structure_code` column to this table in sprint-2. Let's discuss this and come up with a plan to share or not share this table.

lauraGgit replied:
@lpendem please make a story to address it.

lpendem replied:
@lauraGgit story created https://github.com/nciinc/fs-intake-module/issues/79

lauraGgit commented on server/dba/seeders/prod/02-species.js:
```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+module.exports = {
```
Why are species and their notes two separate tables, is there a scenario in which they will be accessed separately?

lpendem replied:
@lauraGgit species and speciesNotes are oneToMany. There aren't accessed separately though.

lpendem replied:
Also, we might create interface to change the species notes.

lauraGgit replied:
seems like we could have them as one until the use case suggested that we needed them sepated.

lauraGgit commented on server/dba/seeders/prod/03-speciesNotes.js:
```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+module.exports = {
```
See above about asking why these are separated

lauraGgit commented on server/dba/seeders/prod/04-forestSpecies.js:
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+module.exports = {
```
I believe in sequelize, you can specify 1:many relationships, but I have not seen that. Is that something that we should specify?

lpendem replied:
new story (# 68) created to handle this.

lauraGgit commented on server/dba/seeders/prod/05-forestLocations.js:
```diff
@@ -0,0 +1,53 @@
+module.exports = {
+  up: function(queryInterface, Sequelize) {
+    let forestLocations = [
+      { id: 1,  forest_id: 3, district: null, allowed: false, type: 'area', description: 'Wilderness' },
```
Have we conducted usability testing on how these are written out? each area is written differently. Let's at least by consistent about whether it says `Road 50` vs `45, 4540` and lets just add an optional `alias` parameter for the human readable names.

lpendem replied:
@lauraGgit Amber gave us the data, but we'll update according to the usability test results.

lauraGgit replied:
@lpendem please go ahead and make a story to update that.

lpendem replied:
@lauraGgit story created: https://github.com/nciinc/fs-intake-module/issues/80

lauraGgit commented on server/docs/api/swagger.yml:
```diff
@@ -172,6 +172,28 @@ paths:
           description: "Invalid input or other error"
         404:
           description: "Application not found"
+  /forests/{id}/regulations:
```
isn't there a `forests` endpoint that was also created?

lauraGgit replied:
@lpendem which story?

sadlerw replied:
it was done with #5

lauraGgit commented on server/package.json:
```diff
@@ -54,25 +54,26 @@
-    "report-coverage":
-      "CODECOV_TOKEN='e61f7ed1-686a-4f48-95b3-02fa7085995e' ./node_modules/nyc/bin/nyc.js report --reporter=lcov > coverage.lcov && codecov",
-    "seed": "sequelize db:seed:all",
+    "report-coverage": "CODECOV_TOKEN='e61f7ed1-686a-4f48-95b3-02fa7085995e' ./node_modules/nyc/bin/nyc.js report --reporter=lcov > coverage.lcov && codecov",
```
for open repos you do not need a token. Please remove this key.

lpendem replied:
@lauraGgit codecov reports are only for server code and they do not match either. We plan to drop this and continue to use the coverage reports from CircleCI artifacts.

lauraGgit commented on server/server-setup.sh:
```diff
@@ -6,4 +6,6 @@ sleep 10
 
```
Let's add a comment for this file that it is used by the Dockerfile.

lauraGgit commented on server/src/controllers/christmas-tree.es6:
```diff
@@ -0,0 +1,92 @@
+    forest: {
+      id: input.id,
+      forestName: input.forestName,
+      description: input.description,
```
It's not clear that this is being used on the front end, should we remove?

sadlerw replied:
being used in sprint-2

lauraGgit commented on server/src/models/forest-locations.es6:
```diff
@@ -0,0 +1,56 @@
+  dialect: url.parse(process.env.DATABASE_URL, true).protocol.split(':')[0]
+};
+
+if (
```
why are these being defined on each model? this seems like this should be specified at runtime or in the sequeliz.rc, and not at the individual model level.

lpendem replied:
@lauraGgit we'll research and refactor this.

lauraGgit replied:
@lpendem could you please let me know which story was created to address this?

lpendem replied:
@lauraGgit fixed this in sorint-2 story https://github.com/nciinc/fs-intake-module/issues/6

lauraGgit commented on server/src/models/forest-regulations.es6:
```diff
@@ -0,0 +1,39 @@
+  dialect: url.parse(process.env.DATABASE_URL, true).protocol.split(':')[0]
+};
+
+if (
```
Same as other model comment.

lauraGgit commented on server/src/models/forests.es6:
```diff
@@ -0,0 +1,80 @@
+const sequelize = new Sequelize(process.env.DATABASE_URL, sequelizeOptions);
+
+module.exports = sequelize.define(
+  'forests',
```
Let's make this forest model, christmas tree specific.

lpendem replied:
We need to this discuss this further

lauraGgit replied:
For the time being let's create a story @lpendem

lpendem replied:
@lauraGgit story: https://github.com/nciinc/fs-intake-module/issues/79

lauraGgit commented on server/src/models/species.es6:
```diff
@@ -0,0 +1,58 @@
+      field: 'name',
+      allowNull: false
+    },
+    webUrl: {
```
is this for the photo? If so let's rename this field `photoUrl`

lpendem replied:
@lauraGgit They are plant profiles (typically PDF links) from USDA. They aren't used on the front-end currently.

lauraGgit replied:
@lpendem for accessibility reasons, PDF links don't seem like a great idea. If they are not being used in the front-end we should remove this.

sadlerw replied:
removed in sprint-2

lauraGgit commented on server/src/models/species.es6:
```diff
@@ -0,0 +1,58 @@
+      type: Sequelize.STRING,
+      field: 'web_url'
+    },
+    photos: {
```
I think we should just have the url and then commit the photos into the repository as a static asset.

lpendem replied:
`photos` is not necessary since we are using file name is same as species name, removing this column as a task in sprint-2

lauraGgit replied:
@lpendem could you please drop a link to which story this task is being added to?

sadlerw replied:
I think it was part of the PR for #4

sadlerw replied:
https://github.com/nciinc/fs-intake-module/blob/sprint-2-development/server/dba/migrations/29-alter-species.js

lauraGgit commented on server/test/christmas-tree-controllers.es6:
```diff
@@ -0,0 +1,101 @@
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(function(res){
+          expect(typeof res.body.forest.species).to.not.equal('undefined');
```
These are pretty vague expectations, let's make them more robust than `non undefined` all the way down.

lpendem replied:
@lauraGgit we have added this task in a sprint-2 story

lauraGgit replied:
@lpendem as a task to which story?

sadlerw replied:
https://github.com/nciinc/fs-intake-module/blob/sprint-2-development/frontend/e2e/xmas-tree-species.e2e-spec.ts

lpendem replied:
@lauraGgit sorint-2 story https://github.com/nciinc/fs-intake-module/issues/5

lpendem commented on circle.yml:
```diff
@@ -30,6 +30,7 @@ test:
     - cd server; yarn run snyk-test
     - cd server; yarn migrate
+    - cd server; yarn seed-ci
     - cd server; yarn seed
```
@lauraGgit forest related data is inserted using `seed`, which currently resets during the build. Whereas `seed-ci` has non-commercial application needed for tests.

lpendem commented on frontend/.angular-cli.json:
```diff
@@ -20,8 +20,9 @@
-        "staging": "environments/environment.staging.ts",
-        "prod": "environments/environment.prod.ts"
+        "prod": "environments/environment.prod.ts",
+        "trees": "environments/environment.trees.ts"
```
@lauraGgit this got corrupted during merge conflict resolution, will make the change.

lauraGgit replied:
@lpendem please link to the commit or the story that is created to address that here.

lpendem commented on frontend/e2e/application-temp-outfitters.e2e-spec.ts:
```diff
@@ -2,13 +2,6 @@ import { TempOutfittersForm } from './app.po';
-    expect<any>(element(by.id(`nav-${section}`)).getAttribute('class')).toMatch('usa-current');
-  });
-};
-
```
@sethalt Looks like this has been removed in the intake side as well.

lpendem commented on frontend/e2e/field-validation.po.ts:
```diff
@@ -91,19 +93,23 @@ export class FieldValidation {
 
     it('should display an error message if the file is not a valid type', () => {
       input.sendKeys(testErrorFile);
+      browser.sleep(500);
```
@sethalt Can you answer @lauraGgit's question?

sethalt replied:
@lauraGgit I will look into this when I do my refactoring on the intake side. We were trying to use browser.wait to solve some of our circle failures, but the problem was that the expected conditions weren't being met so it made things worse.

lpendem commented on frontend/package.json:
```diff
@@ -4,18 +4,20 @@
-    "e2e:ci": "ng e2e --port=4200 --watch=false --progress=false",
-    "e2e": "ng e2e --port=4200",
+    "e2e:ci": "ng e2e --port=4200 -prod -e dev --watch=false --progress=false",
+    "e2e": "ng e2e -prod -e dev --port=4200",
```
@lauraGgit for local development, we need to watch test results during the run.

lauraGgit replied:
ah makes sense.

lpendem commented on frontend/src/app/trees/_mocks/forest.ts:
```diff
@@ -0,0 +1,130 @@
+    { id: 28, district: null, allowed: true, type: 'district', description: 'Zigzag' },
+    { id: 29, district: null, allowed: true, type: 'district', description: 'Hood' },
+    { id: 30, district: null, allowed: true, type: 'district', description: 'Barlow' },
+    { id: 31, district: 'Barlow and Hood', allowed: false, type: 'road', description: 'Highway to hell' },
```
changing this one.

lpendem commented on frontend/src/app/trees/_services/trees.service.ts:
```diff
@@ -0,0 +1,61 @@
+      .catch(this.handleError);
+  }
+
+  private handleError(error: Response | any) {
```
We have a separate story created for error handling #45

lpendem commented on frontend/src/app/trees/forests/forest-resolver.service.ts:
```diff
@@ -0,0 +1,26 @@
+export class ForestResolver implements Resolve<any> {
+  constructor(private service: TreesService, private router: Router) {}
+
+  resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<any> {
```
Resolve interface requires `ActivatedRouteSnapshot`

lpendem commented on frontend/src/app/app-routing.module.ts:
```diff
@@ -68,6 +70,14 @@ const appRoutes: Routes = [
+    resolve: {
+      forest: ForestResolver
+    },
+    data: { title: 'Christmas tree info' }
```
@lauraGgit changed this to `Christmas Tree Permit Guidelines`, and we added a new story(#66) to make this dynamic to each forest.

lpendem commented on frontend/src/app/trees/forests/tree-guidelines/sidebar-view.component.html:
```diff
@@ -0,0 +1,103 @@
+</div>
+<section id="sidebar-page" class="usa-section usa-content">
+
+  <div class="usa-grid">
```
Yes, we are waiting for the scrolling indicators from other project. Story # 67 has been created.

lauraGgit commented on frontend/src/app/trees/forests/tree-guidelines/contact-info/contact-info.component.html:
```diff
@@ -0,0 +1,18 @@
+<div>
```
Let's not have code that we are not using in the repository unless we agree that we need it ahead of time.

sadlerw replied:
we needed this for usability testing of the sidebar - it has been removed now

lpendem commented:
@lauraGgit We have implemented some of these changes in PR (https://github.com/nciinc/fs-intake-module/pull/70/commits) and added stories/tasks for the remainder of the fixes.
